x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtTestCase.java

```diff
@@ -328,15 +378,21 @@ public static OctetSequenceKey randomJwkHmacOidc(final JWSAlgorithm jwsAlgorithm
      * @return HMAC key with UTF-8 bytes, making the key bytes compatible with OIDC UTF-8 string encoding.
      */
     public static OctetSequenceKey conditionJwkHmacForOidc(final OctetSequenceKey hmacKey) {
-        final String bytesAsBase64 = hmacKey.getKeyValue().toString();
-        final byte[] base64AsUtf8 = bytesAsBase64.getBytes(StandardCharsets.UTF_8);
-        final OctetSequenceKey.Builder utf8HmacKeyBuilder = new OctetSequenceKey.Builder(base64AsUtf8);
-        utf8HmacKeyBuilder.keyID(hmacKey.getKeyID()); // Copy null attribute is OK (no-op)
-        utf8HmacKeyBuilder.algorithm(hmacKey.getAlgorithm());
-        utf8HmacKeyBuilder.keyUse(hmacKey.getKeyUse());
-        utf8HmacKeyBuilder.keyOperations(hmacKey.getKeyOperations());
-        utf8HmacKeyBuilder.keyStore(hmacKey.getKeyStore());
-        return utf8HmacKeyBuilder.build();
+        final String passwordKey;
+        if (randomBoolean()) {
+            final Base64URL hmacKeyBytesBase64 = hmacKey.getKeyValue(); // Random bytes => 8 bits/byte search space
```

Comment:
I previously used Base64 of random bytes as a new plain text string. It worked, but gave misleading impressing base64 was required. It is not, so I added some fuzzing to ensure random letter passwords can be used too.